File: Raven.Client.Lightweight/RavenQueryHighlightings.cs
// -----------------------------------------------------------------------
//  <copyright file="RavenQueryHighlightings.cs" company="Hibernating Rhinos LTD">
//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
//  </copyright>
// -----------------------------------------------------------------------

using System.Collections.Generic;
using Raven.Abstractions.Data;

namespace Raven.Client
{
	public class RavenQueryHighlightings
	{
		private readonly List<FieldHighlightings> fields = new List<FieldHighlightings>();

		internal FieldHighlightings AddField(string fieldName)
		{
			var fieldHighlightings = new FieldHighlightings(fieldName);
			this.fields.Add(fieldHighlightings);
			return fieldHighlightings;
		}

		internal void Update(QueryResult queryResult)
		{
			foreach (var fieldHighlightings in this.fields)
				fieldHighlightings.Update(queryResult);
		}
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/RavenQueryHighlightings.java
package net.ravendb.client;

import java.util.ArrayList;
import java.util.List;

import net.ravendb.abstractions.data.QueryResult;


public class RavenQueryHighlightings {
  private final List<FieldHighlightings> fields = new ArrayList<>();

  public FieldHighlightings addField(String fieldName) {
    FieldHighlightings fieldHighlightings = new FieldHighlightings(fieldName);
    this.fields.add(fieldHighlightings);
    return fieldHighlightings;
  }

  public void update(QueryResult queryResult) {
    for (FieldHighlightings fieldHighlightings: this.fields) {
      fieldHighlightings.update(queryResult);
    }
  }

}
